packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1583,6 +1583,21 @@ export class Domain extends DomainBase implements IDomain {
         accessPolicies: accessPolicyStatements,
       });
 
+      if (props.encryptionAtRest?.kmsKey) {
```

Comment:
Following up on my previous comment, what happens if a kms key is not provided? from the docs:

> Specifies whether the domain should encrypt data at rest, and if so, the AWS Key Management Service (KMS) key to use. Can be used only to create a new domain, not update an existing one

This reads that if `EncryptionAtRestOptions` is provided with `Enabled ` set to true and no `KmsKeyId ` is provided - the creation will fail? If so should we create a default kms key if  `encryptionAtRest` is provided? 
From looking at the code it seems that there is no error thrown (at build time) if `EncryptionAtRestOptions` is provided and no key is passed, it will simply be created without a key:

```ts
      encryptionAtRestOptions: {
        enabled: encryptionAtRestEnabled,
        kmsKeyId: encryptionAtRestEnabled
          ? props.encryptionAtRest?.kmsKey?.keyId
          : undefined,
      }, 
```

Reply:
> This reads that if EncryptionAtRestOptions is provided with Enabled set to true and no KmsKeyId is provided - the creation will fail?

In this case the `aws/es` managed key will be used. Which doesn't require any custom permissions.
It did make me realize we didn't actually test it, so i've added `accessPolicies` to the original integ test to make sure the custom resource is used and invoked successfully.

For the custom kms key i've added a separate integ test, so now both those scenarios are covered.